### ISSUE TITLE
Void Trading not working

### ISSUE THREAD

**Issue by sjavi4** [pl3xgaming/Purpur#1054]
### Spark link

-

### Expected behavior

When trading with villager in an unloaded chunk, the number of trading will not be counted.

### Observed/Actual behavior

Villager still counts the number of trading even if the chunk that the villager stay is unloaded.

### Steps/models to reproduce

trade with villager in outer island using the end gateway exploit

### Purpur version

 version
[16:19:40 INFO]: Checking version, please wait...
[16:19:40 INFO]: Current: git-Purpur-1632 (MC: 1.18.2)*
* You are running the latest version

### Agreements

- [X] I am running the latest version of Purpur available from https://purpurmc.org/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Purpur.

**Comment by SzczurekYT** [comment]
@BillyGalbreath hello
Don't close this, as this issue also affects 1.19 the latest, supported version.
Edit: I have allow void trading config enabled so that not the case, I'm testing with this farm: https://youtu.be/FGFoCfRaljM

**Comment by encode42** [comment]
This has been discussed before.

[Here](https://discord.com/channels/685683385313919172/685683819608670209/986697120969015357) and [here](https://discord.com/channels/685683385313919172/685683819608670209/986712979716313218). ([Purpur's support server](https://purpurmc.org/discord))

This is not a simple toggle *or* un-patch. As such, it will not be considered for Purpur.

**Comment by BillyGalbreath** [comment]
I'm actually considering removing the void-trading option altogether since no one is reporting it to the source of the problem, Spigot. Until they fix whatever they broke, Paper's "fix" for the exploit does nothing, which in turns means Purpur's override for Paper's "fix" doesnt do anything at all.

**Comment by SzczurekYT** [comment]
> I'm actually considering removing the void-trading option altogether since no one is reporting it to the source of the problem, Spigot. Until they fix whatever they broke, Paper's "fix" for the exploit does nothing, which in turns means Purpur's override for Paper's "fix" doesnt do anything at all.

Oh didnt't knew that. Then maybe I'm gonna report that. It would help If knew where in the code the bug actually is. Maybe I will try to look at spigot / paper sources, but void trading is more a bug / unintended feature so finding a bug that breaks another bug can be a problem I think.

**Comment by SzczurekYT** [comment]
(From discord)
> best bet at this point in time is to create a bug report on spigot's jira about it. idk how md_5 feels about supporting vanilla bugs, though. he may or may not fix it.

Yeah that's right, not everyone likes bugs. However players shoud have a choice if they want these bugs or not. That's what I don't like about paper - it *fixes/changes* many things without lea ving us anu choice, that's why I came here. It would be best if I could move to fabric (I have small private smp), but fabric doesn't have that much serverside mods, and I'm not good enough at fabric right now to make my own ones. Spigot's API is just so easy and convenient to use.

There is nothing wrong with asking md_5 and other spigot devs what they think about that, so I'm gonna create a bug report, but prior to that I want to do more testing, to see for example if this works on 1.18. In the description of the video I linked before ian said:
> This build is not compatible with many modified servers, such as PAPER and SPIGOT.
So there is a chance that it was broken from the start of this mechanic. ( Video title says it's 1.17+)

**Comment by SzczurekYT** [comment]
Oops, ma farm was built a bit incorrectly, but still I'm pretty sure that the issue is real. It's a bit messy, for example I didn't expected that but on Spigot 1.18 this works. I'm gonna make a lot of tests on spigot, paper and purpur and we will see.

**Comment by SzczurekYT** [comment]
> I'm actually considering removing the void-trading option altogether since no one is reporting it to the source of the problem, Spigot. Until they fix whatever they broke, Paper's "fix" for the exploit does nothing, which in turns means Purpur's override for Paper's "fix" doesnt do anything at all.

Hey @BillyGalbreath 
As I said I made some tests. I tried void trading on spigot, paper and purpur with allow void trading set to true. For each try they were these same settings, this same world, just a differnt server jar. Here are the results:
![image](https://user-images.githubusercontent.com/67533827/180615823-4e0320df-99b8-4ca8-8c3b-91dd47444685.png)
As you can see it's not a spigot's fault. Both on 1.18.2 and on 1.19 the void trading works perfectly fine. Paper breaks it, and purpur fails to fix it.
What do you think about this situation?

**Comment by SzczurekYT** [comment]
@sjavi4  Hey, after some talking in the discord we found the issue: You also need to change ```delay-chunk-unloads-by``` in paper.yml to ```0s``` and this should help.

**Comment by BillyGalbreath** [comment]
Ok, so turns out when I was testing this last time, I had the villager inside the spawn chunks. This is why I could not get it to work on Spigot, and why I assumed it was Spigot's fault.

But thanks to @SzczurekYT and their persistence, I looked into this again and figured out my test was flawed and corrected it. I was then able to get it to work on 1.19 Spigot.

This also made me think about Paper's `delay-chunk-unloads-by` setting, so I set that to `0s` and tried again on Purpur with the `allow-void-trading` set to `true` and lo and behold, it worked.

So yeah, just set the unload delay to 0s for your the world you want to allow void trading in. Either that, or wait the 10s (or whatever you have set) for the chunks to unload before trading with the villager.

[![Purpur #1054](https://i9.ytimg.com/vi/Y1GDoAjyqww/mq3.jpg?sqp=CMCd9JYG&rs=AOn4CLDimNB3HaHtR1cvXk1_GVJUSTduBw)](https://youtu.be/Y1GDoAjyqww)
